File: main/contrib/Windows API Code Pack 1.1/source/Samples/DirectX/VB/Direct2D/Paint2D/RoundRectangleShape.vb
' Copyright (c) Microsoft Corporation.  All rights reserved.


Imports Microsoft.VisualBasic
Imports System
Imports Microsoft.WindowsAPICodePack.DirectX.Direct2D1

Namespace D2DPaint
	Friend Class RoundRectangleShape
		Inherits DrawingShape
		Friend _rect As RoundedRect
		Friend _strokeWidth As Single
		Friend _selectedBrushIndex As Integer


		Friend Sub New(ByVal parent As Paint2DForm, ByVal rect As RoundedRect, ByVal strokeWidth As Single, ByVal selectedBrush As Integer, ByVal fill As Boolean)
			MyBase.New(parent, fill)
			_rect = rect
			_strokeWidth = strokeWidth
			_selectedBrushIndex = selectedBrush
		End Sub

		Protected Friend Overrides Sub Draw(ByVal renderTarget As RenderTarget)
			If _fill Then
				renderTarget.FillRoundedRectangle(_rect, _parent.brushes(_selectedBrushIndex))
			Else
				renderTarget.DrawRoundedRectangle(_rect, _parent.brushes(_selectedBrushIndex), _strokeWidth)
			End If
		End Sub

		Protected Friend Overrides WriteOnly Property EndPoint() As Point2F
			Set(ByVal value As Point2F)
                _rect.Rect = New RectF(_rect.Rect.Left, _rect.Rect.Top, value.X, value.Y)
			End Set
		End Property
	End Class
End Namespace

File: main/contrib/Windows API Code Pack 1.1/source/Samples/DirectX/CS/Direct2D/Paint2D/RoundRectangleShape.cs
// Copyright (c) Microsoft Corporation.  All rights reserved.

using Microsoft.WindowsAPICodePack.DirectX.Direct2D1;

namespace D2DPaint
{
    internal class RoundRectangleShape : DrawingShape
    {
        internal RoundedRect _rect;
        internal float _strokeWidth;
        internal int _selectedBrushIndex;


        internal RoundRectangleShape(Paint2DForm parent, RoundedRect rect, float strokeWidth, int selectedBrush, bool fill)
            : base(parent, fill)
        {
            _rect = rect;
            _strokeWidth = strokeWidth;
            _selectedBrushIndex = selectedBrush;
        }

        protected internal override void Draw(RenderTarget renderTarget)
        {
            if (_fill)
            {
                renderTarget.FillRoundedRectangle(_rect, _parent.brushes[_selectedBrushIndex]);
            }
            else
            {
                renderTarget.DrawRoundedRectangle(_rect, _parent.brushes[_selectedBrushIndex], _strokeWidth);
            }
        }

        protected internal override Point2F EndPoint
        {
            set
            {
                _rect.Rect = new RectF(_rect.Rect.Left, _rect.Rect.Top, value.X, value.Y);
            }
        }
    }
}